subprojects/core/src/main/java/org/gradle/plugin/management/internal/DefaultPluginRequest.java

```diff
@@ -25,49 +25,74 @@
 
 public class DefaultPluginRequest implements PluginRequestInternal {
```

Comment:
Thanks. We'll look into that.

Reply:
I'm not sure this would simplify serialization. We'll still want a single `PluginRequestSerializer` to be used in the Groovy `ScriptPluginFactory`. With split implementation it would have to rely on casting and still do null checks for some properties.

Reply:
> With split implementation it would have to rely on casting 

This particular use of casting feels a lot like pattern matching against the different request variants so I would be ok with it from a design standpoint. From a performance standpoint I assume the implications of a downcast on such a shallow hierarchy are negligible. What would be your concern?

> and still do null checks for some properties

In the pattern matching / variant type interpretation, each branch would only have to check the nullability of the properties that make sense on the particular variant (`ScriptPluginRequest` or `BinaryPluginRequest`), right?